src/codegen.cpp

```diff
@@ -8727,6 +8727,9 @@ static void init(CodeGen *g) {
         // Be aware of https://github.com/ziglang/zig/issues/3275
         target_specific_cpu_args = "";
         target_specific_features = riscv_default_features;
+    } else if (g->zig_target->arch == ZigLLVM_x86) {
+        target_specific_cpu_args = "pentium4";
```

Comment:
can you explain this decision with a comment? There's a high risk of this behavior regressing if it's not clear why this is done.

Reply:
It's because we're really targeting `i686` rather than `i386`, it's pretty much impossible to use many of the language features such as fp16 if you stick use the `x87` only.

This is also what clang uses as base cpu.

Reply:
Thanks, I'll add that as a comment and push to master.